src/handlers/source.ts

```diff
@@ -0,0 +1,57 @@
+import { SourceState } from "../types"
+
+export async function ensureSource(id: string) {
+  const source = await SourceState.get(id)
+  if (source) {
+    return source
+  }
+
+  const initSource = SourceState.create({
+    id,
+    numberOfSessions: 0,
+    heartBeatCounters: [],
+  })
+  await initSource.save()
+  return initSource
+}
+
```

Comment:
Given the usage of these functions, please add reasonable comments/documentation.